packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts

```diff
@@ -0,0 +1,229 @@
+              'Raw',
+              datamodel.Value('Id', datamodel.IdBox('DomainId', sugar.domainId(DOMAIN_NAME))),
+            ),
+          }),
+        ),
+      )
+    })
+
+    await allure.step(ctx, 'The domain is registered', async () => {
+      result!.as('Ok').batch.enum.as('Identifiable').enum.as('Domain')
+
+      final = result!.__c[0].batch.enum.__c[0].enum.__c[0].id.name
+      expect(final).toEqual(DOMAIN_NAME)
+    })
+  })
+})
+
+beforeEach(async (ctx) => {
+  await allure.owner(ctx, 'dulger')
+  await allure.label(ctx, 'permission', 'no_permission_required')
+  await allure.label(ctx, 'sdk', 'Java Script')
+})
+
+test('Register asset with too long type', async (ctx) => {
+  await allure.allureId(ctx, '3739')
+  await allure.label(ctx, 'feature', 'Assets')
+  await allure.label(ctx, 'sdk_test_id', 'register_asset_with_too_long_type')
+  await allure.label(ctx, 'story', 'Account registers an asset with too long type')
+
+  const { client, pre, getBlocksListener } = clientFactory()
+  const blocks = await getBlocksListener()
+
+  const normalAssetDefinitionId = sugar.assetDefinitionId('xor', 'wonderland')
+
+  let tooLongAssetName
+  let invalidAssetDefinitionId
+
+  await allure.step(ctx, 'Account registers an asset with long asset name', async () => {
+    tooLongAssetName = '0'.repeat(2 ** 14)
+    invalidAssetDefinitionId = sugar.assetDefinitionId(tooLongAssetName, 'wonderland')
+  })
+
+  await allure.step(ctx, 'The asset with long name is rejected', async () => {
+    await blocks.wait(async () => {
+      await Promise.all(
+        // we should register these assets as separate transactions, because the invalid
+        // one will be rejected
+        [normalAssetDefinitionId, invalidAssetDefinitionId!].map(async (id) => {
+          await client.submitExecutable(
+            pre,
+
+            pipe(
+              sugar.identifiable.newAssetDefinition(id, datamodel.AssetValueType('BigQuantity')),
+              sugar.instruction.register,
+              sugar.executable.instructions,
+            ),
+          )
+        }),
+      )
+    })
+  })
+  await allure.step(ctx, 'The normal asset is registered', async () => {
+    const queryResult = await client.requestWithQueryBox(pre, sugar.find.allAssetsDefinitions())
+
+    const existingDefinitions: datamodel.AssetDefinitionId[] = queryResult
+      .as('Ok')
+      .batch.enum.as('Vec')
+      .map((val) => val.enum.as('Identifiable').enum.as('AssetDefinition').id)
+
+    expect(existingDefinitions).toContainEqual(normalAssetDefinitionId)
+    expect(existingDefinitions).not.toContainEqual(invalidAssetDefinitionId!)
+  })
+})
+
+test('Register an account with long account name', async (ctx) => {
+  await allure.allureId(ctx, '4077')
+  await allure.label(ctx, 'feature', 'Account')
+  await allure.label(ctx, 'sdk_test_id', 'register_account_with_long_account_name')
+  await allure.label(ctx, 'story', 'Register an account with long name')
```

Comment:
I don't really understand the value of this suggestion. All annotations that we can put in advance, we put before each. If we can't annotate in advance, it's better to mark up each specific test with the necessary information for readability.

Reply:
It is called __domain-specific language__.

`allure` utility is general, not specific to your domain. It doesn't know that you are using labels `feature`, `sdk_test_id`, `story`. It doesn't know that `allureId` is supposed to be a number, so that you have to pass it as a string instead.

Here I see it useful to define a small domain-specific function that actually _defines_ your testing domain, known attributes, __statically__. The value is that you are less likely to make a mistake/typo, you can have a hint from type system, you repeat less. Instead of writing time and time again this:

```ts
await allure.allureId(ctx, '4077')  
await allure.label(ctx, 'feature', 'Account')  
await allure.label(ctx, 'sdk_test_id', 'register_account_with_long_account_name')
await allure.label(ctx, 'story', 'Register an account with long name')
```

You can write this:

```ts
await annotate(ctx, {
    // expects a number
    allureId: 4077,

    feature: 'Account',
    stdTestId: 'register_account_with_long_account_name',
    story: 'Register an account with long name',
    
    // Ooops, typo! Get a hint and a type error
    feeture: 'Bad'
})
// internally, this function will call `allure` for you
```

And using this function that _knows_ your domain is a more robust approach. In addition, it's just less boilerplate and is arguably more expressive, as it doesn't contain extra noise.

I wouldn't say that reducing code repetition and adding layers of abstraction is always a good idea, but in this case, it is clearly (for me) worth it.